File: src/main/resources/static/js/_1_domain/tag.js
/* _1_domain/tag.js — Tag entity. Pure. */
(function () {
  function normalize(raw) {
    if (!raw) return null;
    return {
      id:    raw.id,
      name:  raw.name || '',
      color: raw.color || null,
    };
  }

  function hasColor(t) { return !!(t && t.color); }

  window.Domain = window.Domain || {};
  window.Domain.Tag = {
    normalize: normalize,
    hasColor: hasColor,
  };
})();

File: br-application/src/main/java/br/cdb/feature/f004/_0_domain/model/Tag.java
package br.cdb.feature.f004._0_domain.model;

import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.UUID;

@NullMarked
public record Tag(
        UUID id,
        UUID personId,
        String name,
        String color,
        @Nullable LocalDateTime createdAt
) {}
